Automattic/wp-calypso
# Backup: Fix the upsell logic to consider retention on backup dashboard

Related to 1203958262767124-as-1204768921350320
Issue: https://github.com/Automattic/jpop-issues/issues/8334

## Proposed Changes

As reported in the issue, we have inconsistencies while suggesting the storage upgrade on the backup dashboard (incorrect) and the retention settings page (correct). This is due to the fact that we were not considering the current retention of the plan while computing the recommended upgrade on the backup dashboard, on the other hand, the retention settings page does consider the current retention and hence shows the correct recommendation.

The changes:
- Fix the logic to consider current retention (custom/default as per the plan), the additional storage estimation buffer.
- Make the retention settings page use the same function for upsell recommendation and **delete** the repeated code (when compared with the fixed one!)

## Testing Instructions



1) Make sure you have a site, which requires a storage add-on upgrade as per the current site size and usage. Or you can make a site in such a state by updating the site storage cache, for e.g.:

```
vpshell > vp_cache_set("site_last_backup_size_1377093", 2147483648);  // set site size to 2GB
vpshell > vp_cache_set("site_used_storage_1377093", 9663676416); // set total usage to 9GB out of 10GB plan
```
2) Now check the retention setting page using the generated Jetpack Cloud/Calypso link for your site. The recommendation will show that we need a 100GB upgrade for 30-days retention(Default) if you have used the above values to update the site's storage cache.
3) Cross-check the recommendation shown on the backup dashboard with the one on the retention settings page, they should be the same. To confirm or simulate the issue you can compare the two recommendations on https://cloud.jetpack.com/ - it will show the inconsistency.

#### The issue: 

On the retention page, it shows 100GB needed:

<img width="696" alt="Screenshot 2023-07-13 at 6 19 28 PM" src="https://github.com/Automattic/wp-calypso/assets/13975226/449e458d-80ed-4b7e-b35e-d5da1e81122e">

But on upsell banner on the dashboard page, it shows 10GB is needed:

<img width="777" alt="Screenshot 2023-07-13 at 6 19 37 PM" src="https://github.com/Automattic/wp-calypso/assets/13975226/3209cf42-6b3e-41af-afae-f60785f15c70">

After this bugfix, the two places should show the same recommendations.
## Pre-merge Checklist
